compiler-core/src/type_/tests.rs

```diff
@@ -284,7 +287,16 @@ fn compile_statement_sequence(src: &str) -> Result<Vec1<TypedStatement>, crate::
             has_javascript_external: false,
         },
     )
-    .infer_statements(ast)
+    .infer_statements(ast, errors);
+    match (res, Vec1::try_from_vec(errors.to_vec())) {
+        (Ok(res), Err(_)) => Ok(res),
+        (Ok(_), Ok(errors)) => Err(errors),
+        (Err(err), Ok(mut errors)) => {
+            errors.push(err);
+            Err(errors)
+        }
+        (Err(err), Err(_)) => Err(Vec1::new(err)),
+    }
```

Comment:
What is this code doing?

Reply:
There were a couple where we were directly instantiating and calling the exprtyper instead of going through the analysis module. In those tests we were only looking at the return value. This meant that the errors appended to the errors vector were ignores. This bit here is appending the errors vector with the return value so that all errors are accounted for in the returned error